tools/codegen/dest/lazy_ir.py

```diff
@@ -246,19 +304,20 @@ class GenLazyShapeInferenceDefinition:
     tensor_class: str
 
     @method_with_native_function
+    # def gen_lazy_shape_inference_decl(f: NativeFunction, backend_index: BackendIndex, tensor_class: str) -> List[str]:
     def __call__(self, f: NativeFunction) -> List[str]:
         sig = kernel_signature(f, self.backend_index)
-
-        # Lazy IR stuff
+        metadata = self.backend_index.get_kernel(f)
+        assert metadata is not None
         schema = LazyIrSchema(f.func)
         value_types = schema.filtered_types(values=True, scalars=False)
-        lazy_tensor_decls_str = lazy_tensor_decls(value_types, self.tensor_class)
+        lazy_tensor_decls_str = lazy_tensor_decls(value_types, self.tensor_class, schema)
         node_ctor_input_str = node_ctor_inputs(schema)
 
         # Only generate shape/dtype fn for non-structured kernels,
         # since we just use the meta function for structured kernels
         if not f.structured and f.structured_delegate is None:
-            shape_sig = ComputeShapeSignature(f)
+            shape_sig = ComputeShapeSignature(metadata.kernel, f)
```

Comment:
This doesn't seem like a big deal - but any reason why we need to use the `metadata.kernel` to choose how to name the shape functions? For example, we could do something like `f.func.name.unambiguous_name()` instead

Reply:
I can't answer off the top of my head.  I probably didn't know about unambiguous_name().  I think i needed metadata.kernel to overcome some ambiguity so this may be possible to substitute. I'll have to test it separately (back on staging) and then drop the update in if it works.

Reply:
ok, so here is one case where they differ.

ipdb> p metadata.kernel
'arange_out'
ipdb> p f.func.name.unambiguous_name()
'arange_start_out'

I'm not sure what the right thing to do is.

Reply:
leaving for a refactor sounds fine.

I think at a high level, what we probably want to have happen (based on what some of the other codegen in core does), is that the shape signatures are part of some API (the "lazy API", or "lazy shape API") that knows how to generate its name. That way at both the place where you define those functions in the codegen, and at the place where you call those functions, you can use the same api call to generate the name (and you don't have to remember whether or not it was actually using `metadata.kernel` or `f.unambiguous_name()`.

Reply:
yea that makes sense. i'll reflect that in the other (refactor) issue.